### PR TITLE
Create Stripe Tax Transaction after fully shipping an order

### PR DESCRIPTION
**Pull request by mad-eel**
## Summary by CodeRabbit

- New Features
  - Automatically creates Stripe tax transactions when orders are fully shipped, improving post-shipment tax handling.
- Bug Fixes
  - Ensures tax calculations refresh correctly by incorporating updated line item amounts, reducing stale tax results.
- Changes
  - Moves tax transaction creation from payment authorization to shipment completion for more accurate timing.
- Tests
  - Added coverage for background job execution and shipment-triggered tax transaction behavior, including edge cases.

### CODE REVIEW

File: app/jobs/spree_stripe/create_tax_transaction_job.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module SpreeStripe
+  class CreateTaxTransactionJob < ApplicationJob
```
Comment (damianlegawiec) [inline app/jobs/spree_stripe/create_tax_transaction_job.rb:2]:
```suggestion
  class CreateTaxTransactionJob < BaseJob
```

File: app/models/spree_stripe/shipment_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    private
+
+    def create_tax_transaction_async
+      return unless order.fully_shipped?
+      return if order.stripe_tax_calculation_id.blank?
+
+      payment_intent = order.payment_intents.last
+      return if payment_intent.blank?
+
+      SpreeStripe::CreateTaxTransactionJob.perform_later(order.store_id, payment_intent.stripe_id, order.stripe_tax_calculation_id)
```
[the remark this answers is not in the document]
Comment (mad-eel) [reply app/models/spree_stripe/shipment_decorator.rb:16]: The association exists in `SpreeStripe::OrderDecorator`.
